### PR TITLE
Disable apply button at initial state and move agents to their original position

### PR DESCRIPTION
**Pull request by sortizowlh**
Hi team!

When managing agents into a group generates an error, all the agents move to their original positions.
Also, the `Apply changes` button is disabled when all the agents are in their original position.

**To test it:** 

1. Go to Management/Groups

- **Given** the browser is in wazuh app with more than 1 user available
- **When** the user navigates to Management/Groups
- **Then** the Groups section will be shown

2. Create a Group

- **Given** the Group section
- **When** the user click in add new group
- **And** the user fill the name with "test"
- **And** the user click on "Save new Group"
- **Then** the group will be displayed in Groups table

3. Enter in Manage Agents in Groups section

- **Given** the group section with the group "test" created
- **When** the user click on test
- **And** the user click on "Manage agents" in the Test group section
- **Then** the agents will be displayed in Available agents and current agents in the group will be empty

4. Try to move agents in Manage agents of group test

- **Given** the section to move agent from some groups
- **When** the user can move all agents to the right or left back, save them and move them again
- **Then** the button to save will be enabled if the user move some of the users and there isn't any errors (except when the user will try to move an inactive or never connected agent, that will persist in the left side)

Solves: #3332

### CODE REVIEW

**Review by asteriscos — CHANGES_REQUESTED** [review]
Test: ❌

When one of the agents is added and the rest is not, it doesn't end in the right group column. Also, the `See full error` doesn't do anything. In addition, the number of `added` or `removed` items is not reset after the operation.

Peek 2021-09-15 13-42

File: public/components/management/groups/multiple-agent-selector.tsx [hunk 1]
```diff
@@ -9,707 +9,897 @@
...
-      item.forEach(elem => this.moveItem(elem, from, to, type));
-      this.checkLimit();
-    } else {
-      item = JSON.parse(item);
-      const idx = from.findIndex(x => x.key === item.key);
-      if (idx !== -1) {
-        from.splice(idx, 1);
-        item.type = !item.type ? type : '';
-        to.push(item)
+    moveItem = (item, from, to, type) => {
```
Comment (mpRegalado) [inline public/components/management/groups/multiple-agent-selector.tsx:405]: This functions mutates the state. It doesn't seem to have been changed in this PR but it should be addressed

File: public/components/management/groups/multiple-agent-selector.tsx [hunk 2]
```diff
@@ -9,707 +9,897 @@
...
+        //relaod agent columns
+        this.reload('right');
+        this.reload('left');
+
+        try {
+          //delete nodes selected
+          await WzRequest.apiReq('DELETE', `/agents/group`, {
+            params: {
+              group_id: this.props.currentGroup.name,
+              agents_list: itemsToSave.addedIds.toString(),
```
Comment (mpRegalado) [inline public/components/management/groups/multiple-agent-selector.tsx:362]: Deleting all agents from the `itemsToSave` list upon error means that the agents that caused the error in the first place are not in the group. So attempting to delete them causes another error. The final consequence is simply a console log, but the `failedIds` list should be updated when there is an error and used to filter what items are to be deleted.

**Review by mpRegalado — CHANGES_REQUESTED** [review]
The code performs the function required. Upon encountering an error, it reverts all the changes to its original state.
However, the way that it does so causes further errors to be logged into the console.

File: public/components/management/groups/multiple-agent-selector.tsx [hunk 3]
```diff
@@ -9,707 +9,897 @@
...
+        if (itemsToSave.addedIds.length) {
+          const addResponse = await WzRequest.apiReq('PUT', `/agents/group`, {
+            params: {
+              group_id: this.props.currentGroup.name,
+              agents_list: itemsToSave.addedIds.toString(),
+            },
+          });
+
+          if (addResponse.data.data.failed_ids) {
+            failedIds.push(...addResponse.data.data.failed_ids);
```
Comment (mpRegalado) [inline public/components/management/groups/multiple-agent-selector.tsx:293]: This code is never executed. An error in the assignment of `addResponse` raises an exception that skips this section to the `catch` phase, meaning that `failedIds` is never updated.

**Review by CPAlejandro — APPROVED** [review]
CR and Testing: LGTM!

**Pull request merged**